lib/gc/Dialect/Microkernel/MicrokernelOps.cpp

```diff
@@ -8,7 +8,369 @@
+    if (!isa<TypeAttr>(dataType))
+      return failure();
+    dataTypes.push_back(dataType);
+    return success();
+  };
+  if (parser.parseCommaSeparatedList(parseTypeAttr) || parser.parseRParen())
+    return failure();
+
+  result.addAttribute(DATA_TYPE, builder.getArrayAttr(dataTypes));
+  return success();
+}
+
+template <typename FLAGS>
+static ParseResult parseFlagsImpl(OpAsmParser &parser, OperationState &result,
+                                  const std::string_view &flagsName) {
+  auto &builder = parser.getBuilder();
+  if (parser.parseKeyword(flagsName) || parser.parseEqual() ||
+      parser.parseLParen())
+    return failure();
+
+  SmallVector<Attribute, 4> flags;
+  auto parseFlags = [&]() -> ParseResult {
+    FLAGS flag;
+    if (parseEnum<FLAGS>(flag, parser))
+      return failure();
+    flags.push_back(builder.getI64IntegerAttr(static_cast<int64_t>(flag)));
+    return success();
+  };
+  if (parser.parseCommaSeparatedList(parseFlags) || parser.parseRParen())
+    return failure();
+  result.addAttribute(flagsName, builder.getArrayAttr(flags));
+  return success();
+}
+
+template <typename FLAGS>
+static LogicalResult
+verifyUniquenessAndConsistency(ArrayAttr flags, Operation *op,
+                               const std::string_view &flagsName) {
+  SmallVector<int64_t> flagsAsInt;
+  for (auto flag : flags)
+    flagsAsInt.push_back(cast<IntegerAttr>(flag).getInt());
+
+  // check uniqueness
+  std::sort(flagsAsInt.begin(), flagsAsInt.end());
+  auto *it = std::unique(flagsAsInt.begin(), flagsAsInt.end());
+  if (it != flagsAsInt.end())
+    return op->emitOpError() << "expected " << flagsName << " to be unique";
+  // none flag conflicts with all the others
+  if (llvm::is_contained(flagsAsInt, static_cast<int64_t>(FLAGS::NONE)) &&
+      flagsAsInt.size() != 1) {
+    return op->emitOpError()
+           << "'none' " << flagsName << " conflicts with others";
+  }
+  return success();
+}
+
+/////////////////////////////////////////////////////
+// Start of BrgemmDispatchOp
+
+void BrgemmDispatchOp::print(OpAsmPrinter &printer) {
+  printInputImpl<BrgemmDispatchOp>(printer, *this);
+  auto getOpFlags = [this]() -> ArrayAttr { return this->getFlags(); };
+  printFlagsImpl<BrgemmFlagsAttr>(printer, getOpFlags, FLAGS_NAME);
+  printDataTypeImpl<BrgemmDispatchOp>(printer, *this);
+}
+
+ParseResult BrgemmDispatchOp::parse(OpAsmParser &parser,
+                                    OperationState &result) {
+  if (failed(parseOperandImpl(parser, result)) ||
+      failed(parseFlagsImpl<BrgemmFlags>(parser, result, FLAGS_NAME)))
+    return failure();
+  if (failed(parseDataTypeImpl(parser, result)))
+    return failure();
+  return parser.parseOptionalAttrDict(result.attributes);
+}
+
+static LogicalResult verifyBrgemmDataTypes(ArrayAttr dtypes,
+                                           BrgemmDispatchOp op) {
+  if (dtypes.size() != 2) {
+    return op->emitOpError() << "data types size should be 2";
+  }
```

Comment:
llvm code style conventions
```suggestion
  if (dtypes.size() != 2)
    return op->emitOpError() << "data types size should be 2";
```
not marking other places

Reply:
Checked & changed all brackets in this file per conventions.